### PR TITLE
Classifier 93

### PR DESCRIPTION
**Pull request by kalsaraM**
**Jira Cloud**

1. Connecting Jira flow implemnted here,and below config need to be set (JIRA_API_TOKEN ,JIRA_USERNAME)

      JIRA_API_TOKEN can be create from [here](https://id.atlassian.com/manage-profile/security/api-tokens) 

  ```
  curl -X POST https://{{DOMAIN}}/classifier/integration/jira/cloud/toggle-subscription \
  -H "Authorization: Bearer YOUR_JWT_TOKEN" \
  -d '{
    "isConnect": "true",
  }'  
  ```

2.  Jira Integration for Receiving Ticket Information upon Creation or Label Update via Webhooks flow implemented.

      Webhook need to be create Manually and seceret token and webhookId need to be setup as a config

3. Jira issue label update flow implemnted

```
curl -X POST https://{{DOMAIN}}/classifier/integration/jira/cloud/update\
-H "Authorization: Bearer YOUR_JWT_TOKEN" \
-d '{
    "issueKey": "KAL-1",
    "labels": ["kal-46","kal-45"]
}'
```

**TODO**

    

- [ ] Need to bypass Ruuter Authentication
- [ ] Signature Validation flow issue fix
- [ ] Need to Figure out a flow for WebhookId identification

### CODE REVIEW

File: DSL/DMapper/lib/helpers.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { createHmac, timingSafeEqual } from "crypto";
+
+export function verifySignature(payload, headers) {
+  const signature = headers["x-hub-signature"];
+  const SHARED_SECRET = "wNc6HjKGu3RZXYNXqMTh";
```
Comment (Thirunayan22) [inline DSL/DMapper/lib/helpers.js:5]: Is this a secret key @kalsaraM ? If so we have to add it to a secrets file and refer to it in the environments variable or through docker

File: DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/accept.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
+declaration:
+  call: declare
+  version: 0.1
+  description: "Description placeholder for 'ACEEPT'"
```
Comment (Thirunayan22) [inline DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/accept.yml:4]: Just a small typo here I think @kalsaraM, I guess it should be  "ACCEPT"

File: DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/toggle-subscription.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+subscribe_jira:
+  call: http.put
+  args:
+    url: "[#JIRA_CLOUD_DOMAIN]/rest/webhooks/1.0/webhook/[#JIRA_WEBHOOK_ID]"
+    headers:
+      Authorization: ${auth_header.response.body.val}
+    body:
+      enabled: true
+  result: res
```
Comment (Thirunayan22) [inline DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/toggle-subscription.yml:46]: Any reason we are naming the output as "res" here @kalsaraM, I noticed in the previous endpoints you had names which describes their function like "auth_header".

File: DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/toggle-subscription.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  next: assign_jira_webhook_status
+
```
Comment (Thirunayan22) [inline DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/toggle-subscription.yml:48]: Can we add comment here and in other places where one flow ends @kalsaraM ? What do you think?

File: DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/toggle-subscription.yml (hunk 3 above)
Comment (kalsaraM) [reply DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/toggle-subscription.yml:46]: it just a short name for reponse

File: DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/toggle-subscription.yml (hunk 4 above)
Comment (kalsaraM) [reply DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/toggle-subscription.yml:48]: shall we discuss this on tommorow DSM

Comment (Thirunayan22) [reply DSL/Ruuter.private/DSL/POST/classifier/integration/jira/cloud/toggle-subscription.yml:48]: Sure

**Review by Thirunayan22 — APPROVED** [review]
Good to go 👍

**Pull request merged**